src/llm_vm/completion/optimize.py

```diff
@@ -181,7 +182,16 @@ def __init__(self, storage=local_ephemeral(), MIN_TRAIN_EXS = 1, MAX_TRAIN_EXS =
         self.openai_key = openai_key
         self.data_synthesizer = data_synthesis.DataSynthesis(0.87, 50)
 
-    def complete(self, stable_context, dynamic_prompt, data_synthesis = False, finetune = False, **kwargs):
+    def complete(self, stable_context, dynamic_prompt, data_synthesis = False, finetune = False, regex = None, type = None, choices = None, **kwargs):
+        if regex is not None:
```

Comment:
could this set of ifs be made into a `match` where we have a single param for selecting the type of completion. it could be 
```python 
match completion_sort:
   case "regex":

   case "type"


   ```
etc

Reply:
Sure will do